### PR TITLE
KPIV-17: move member flags under join options

### PR DESCRIPTION
**Pull request by icehess**
This moves member flags under `Join-Options` to make it easier for apps to send join flags for member in conference command without needing to update the code each time we want to add/support new flags.

### CODE REVIEW

File: src/call_cmd/ecallmgr_call_command.erl [hunk 1]
```diff
@@ -887,11 +887,13 @@ maybe_set_nospeak_flags(Node, UUID, JObj) ->
...
                           ,{<<"moderator">>, <<"Moderator">>}
                           ,{<<"vmute">>, <<"Video-Mute">>}
+                          ,{<<"endconf">>, <<"Endconf">>}
                           ]).
 
 -spec get_conference_flags(kz_json:object()) -> binary().
 get_conference_flags(JObj) ->
-    case kz_json:to_proplist(JObj) of
+    JoinOptions = kz_json:get_ne_json_value(<<"Join-Options">>, JObj, kz_json:new()),
+    case kz_json:to_proplist(JoinOptions) of
```
Comment (lazedo) [inline src/call_cmd/ecallmgr_call_command.erl]:
```suggestion
    case kz_json:to_proplist(<<"Join-Options">>, JObj) of
```

File: src/call_cmd/ecallmgr_call_command.erl [hunk 2]
```diff
@@ -887,11 +887,13 @@ maybe_set_nospeak_flags(Node, UUID, JObj) ->
                           ,{<<"deaf">>, <<"Deaf">>}
                           ,{<<"moderator">>, <<"Moderator">>}
                           ,{<<"vmute">>, <<"Video-Mute">>}
+                          ,{<<"endconf">>, <<"Endconf">>}
```
Comment (lazedo) [inline src/call_cmd/ecallmgr_call_command.erl]:
```suggestion
                          ,{<<"endconf">>, <<"End-Conference">>}
```

**Review by lazedo — APPROVED** [review]
lets wait for related pr to merge

File: test/ecallmgr_call_command_tests.erl [hunk 3]
```diff
@@ -15,20 +15,21 @@
 -include_lib("eunit/include/eunit.hrl").
 
 all_conference_flags_test() ->
-    JObj = kz_json:from_list([{<<"Mute">>, 'true'}
+    JObj = conf_join_options([{<<"Mute">>, 'true'}
                              ,{<<"Deaf">>, 'true'}
                              ,{<<"Moderator">>, 'true'}
+                             ,{<<"End-Conference">>, 'true'}
```
Comment (lazedo) [inline test/ecallmgr_call_command_tests.erl:21]: test will always fail ?

**Review by lazedo — CHANGES_REQUESTED** [review]
test seems to always fail

**Comment by icehess** [comment]
@lazedo CI is passing now, anything else is needed for these PRs to be merged?

**Pull request merged**